drivers/wifi/nrf700x/src/wpa_supp_if.c

```diff
@@ -2370,6 +2555,7 @@ int nrf_wifi_wpa_supp_sta_set_flags(void *if_priv, const u8 *addr,
 	ret = 0;
 
 out:
+	k_mutex_unlock(&vif_ctx_zep->vif_lock);
```

Comment:
I can't comment at that particular line, but at line 2529 you jump to `out` w/o locking mutex.